### PR TITLE
Cleanup RBAC for KDD

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Make RBAC docs more clear, and rename the kdd rbac yaml so that it's less confusing with the etcd version.

TODO:
- [x] Add RBAC notes to integration guide

Fixes https://github.com/projectcalico/calico/issues/402

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: master/getting-started/kubernetes/installation/hosted/hosted.md [hunk 1]
```diff
@@ -15,7 +15,15 @@ kubectl apply -f calico.yaml
 
 > **NOTE**
 >
-> Make sure you configure the provided ConfigMap with the location of your etcd cluster before running the above command. 
+> Make sure you configure the provided ConfigMap with the location of your etcd cluster before running the above command.
+
+## RBAC
+
+If using the above manifest on an RBAC enabled cluster, apply the `ClusterRole` and `ClusterRoleBinding` specs as well:
```
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/hosted.md]: We do mention this on the index page of the hosted installation - so should we remove from there?

Also, does this need to be done before applying the calico.yaml?  If so, we should re-order.

File: master/getting-started/kubernetes/installation/integration.md [hunk 2]
```diff
@@ -192,6 +193,20 @@ calico-policy-controller                 1/1       Running   0          1m
...
+APIs.  To do this, subjects and roles must be configured in the Kubernetes API and Calico components must be provided with the appropriate
+tokens or certificates to presnt which identify it as the configured API user.
+
+Detailed instructions for configuring Kubernetes RBAC are outside the scope of this document.  For more information,
+please see the [upstream Kubernetes documentation](https://kubernetes.io/docs/admin/authorization/rbac/) on the topic.
+
+The following yaml files define the necessary API permissions required by Calico (by datastore).
+
+- [etcd datastore](rbac.yaml)
+- [Kubernetes API datastore](hosted/rbac-kdd.yaml)
```
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/integration.md]: Above it says we don't cover KDD in this guide.

Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/integration.md]: lol yeah

**Comment by caseydavenport** [comment]
@robbrockbank ok, updated. PTAL

File: master/getting-started/kubernetes/installation/hosted/index.md [hunk 3]
```diff
@@ -92,14 +92,6 @@ Calico's manifests assign its components one of two service accounts.
 Depending on your cluster's authorization mode, you'll want to back these
 ServiceAccounts with the neccessary permissions.
```
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/index.md]: Is this para still necessary?  As it stands it doesn't really say anything about how to do that.

Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/index.md]: I think it's still useful, and not sure where else it should go right now.

I think we should leave it for now, and potentially re-jig in a future update.

**Pull request merged**